gluten-core/src/main/java/io/glutenproject/substrait/expression/StructLiteralNode.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.substrait.expression;
+
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow;
+
+import io.substrait.proto.Expression;
+import io.substrait.proto.Expression.Literal.Builder;
+import io.glutenproject.substrait.type.TypeNode;
+import io.glutenproject.substrait.type.StructNode;
+
+import java.util.ArrayList;
+
+public class StructLiteralNode extends LiteralNodeWithValue<GenericInternalRow> {
+  public StructLiteralNode(GenericInternalRow row, TypeNode typeNode) {
```

Comment:
Can we remove `BinaryStructNode` and use this instead?

https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/java/io/glutenproject/substrait/expression/BinaryStructNode.java

Reply:
ok.

Reply:
done. Other xxxListNode is removed since they are no longer needed after `ListLiteralNode`